### PR TITLE
No external event

### PR DESCRIPTION
**Pull request by woksin**
Builds on this PR #278. Pull that one inn first.

This PR remove the IExternalEvent interface as well as some exceptions that are not going to be used. It also logs warnings instead of throwing exception when registering event handlers

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1140654762299587/1170646860032197)

### CODE REVIEW

File: Source/Events.Handling/EventHandlerProvider.cs [hunk 1]
```diff
@@ -18,47 +19,80 @@ public class EventHandlerProvider : ICanProvideEventHandlers
...
+        public IEnumerable<AbstractEventHandler> Provide()
+        {
+            var eventHandlers = new List<AbstractEventHandler>();
+            foreach (var eventHandlerType in _eventHandlerTypes)
             {
-                ThrowIfMissingAttributeForEventHandler(type);
-                var eventHandlerId = type.GetCustomAttribute<EventHandlerAttribute>().Id;
-                var eventMethods = type.GetMethods(BindingFlags.Instance | BindingFlags.DeclaredOnly | BindingFlags.Public)
-                                                    .Where(_ => _.Name == AbstractEventHandler.HandleMethodName && TakesExpectedParameters(_));
+                if (!CheckEventHandlerAttributes(eventHandlerType)) break;
```
Comment (einari) [inline Source/Events.Handling/EventHandlerProvider.cs]:
```suggestion
                if (!CheckEventHandlerAttributes(eventHandlerType)) continue;
```

File: Source/Events.Handling/EventHorizon/ExternalEventHandlerProvider.cs [hunk 2]
```diff
@@ -21,76 +20,113 @@ public class ExternalEventHandlerProvider : ICanProvideEventHandlers
...
 
         /// <inheritdoc/>
-        public IEnumerable<AbstractEventHandler> Provide() => _eventHandlerTypes.Select(eventHandlerType =>
+        public IEnumerable<AbstractEventHandler> Provide()
+        {
+            var eventHandlers = new List<AbstractEventHandler>();
+            foreach (var eventHandlerType in _eventHandlerTypes)
             {
-                ThrowIfMissingAttributes(eventHandlerType);
+                if (!CheckEventHandlerAttributes(eventHandlerType)) break;
```
Comment (einari) [inline Source/Events.Handling/EventHorizon/ExternalEventHandlerProvider.cs]:
```suggestion
                if (!CheckEventHandlerAttributes(eventHandlerType)) continue;
```

**Review by einari — CHANGES_REQUESTED** [review]

File: Source/Events.Handling/EventHorizon/ExternalEventHandlerProvider.cs [hunk 3]
```diff
@@ -21,76 +20,113 @@ public class ExternalEventHandlerProvider : ICanProvideEventHandlers
...
+                var eventTypesAndMethods = eventMethods.Select(_ => (_.GetParameters()[0].ParameterType, _));
+                var eventHandlerMethods = new List<EventHandlerMethod<IPublicEvent>>();
+                foreach ((var eventType, var handleMethod) in eventTypesAndMethods)
                 {
-                    (var eventType, var methodInfo) = (eventTypeAndMethod.eventType, eventTypeAndMethod.methodInfo);
-                    ThrowIfEventMissingArtifactAttribute(eventType);
-                    var producerMicroservice = eventType.GetCustomAttribute<ArtifactAttribute>().Artifact;
-                    eventHandlerMethods.Add(new EventHandlerMethod<IExternalEvent>(eventType, methodInfo));
-                });
+                    if (!CheckEventAttributes(eventType)) break;
```
Comment (einari) [inline Source/Events.Handling/EventHorizon/ExternalEventHandlerProvider.cs]: The stream is immutable - so we need to then consider the entire ExternalEventHandler to be 'broken'.
Any events in a handler with event types without the attribute should cause it to be broken.

**Review by einari — CHANGES_REQUESTED** [review]
Immutable EventHandlers

**Review by jakhog — APPROVED** [review]
Nice :)

**Pull request merged**